### PR TITLE
fix: #4176 File Handler on `System` client.

### PR DESCRIPTION
**Pull request by EdwinBetanc0urt**
https://github.com/adempiere/adempiere/assets/20288327/6a509fd8-7531-4533-8727-0ca896fd320d

imagen

#### Additional context
fixes https://github.com/adempiere/adempiere/issues/4176

### CODE REVIEW

**Comment by e-Evolution** [comment]
hi @EdwinBetanc0urt I see the correction as important since I can see a leak resources by not closing the files which indicates that the correction is correct, on the other hand I can see that you are adding the joda-time-2.8.1.jar library which is not used in the fix at least not that I can see, on the other hand using JDK 11 why not use the standard JDK time API instead of using joda?

kind regards
Victor

**Comment by EdwinBetanc0urt** [comment]
Hi @e-Evolution, there is already a version of `joda-time` but it is used by the storefront for the online store.
https://github.com/adempiere/adempiere/blob/develop/tools/lib/storefront/joda-time-2.10.4.jar

However, this version is not compatible with aws-java-sdk-core.
https://github.com/adempiere/adempiere/blob/develop/tools/lib/aws-java-sdk-core-1.11.795.jar

Generating this error:
```
===========> SetupFileStorageSystem.process: java.lang.NoSuchMethodError: 'org.joda.time.format.DateTimeFormatter org.joda.time.format.DateTimeFormatter.withZoneUTC()
```

**Review by e-Evolution — COMMENTED** [review]
Here some comments about your Code

File: base/src/org/compiere/model/MAttachment.java [hunk 1]
```diff
@@ -522,6 +530,7 @@ private boolean saveLOBData() {
 						.saveAttachment();
 				} catch (Exception e) {
 					log.warning("Error saving attachment: " + e.getLocalizedMessage());
+					e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (EdwinBetanc0urt) [reply base/src/org/compiere/model/MAttachment.java]: @e-Evolution the trace printing has already been removed

File: .classpath [hunk 2]
```diff
@@ -314,6 +314,7 @@
 	<classpathentry kind="lib" path="tools/lib/httpcore-4.4.10.jar"/>
 	<classpathentry kind="lib" path="tools/lib/httpmime-4.5.6.jar"/>
 	<classpathentry kind="lib" path="tools/lib/commons-codec-1.6.jar"/>
+	<classpathentry kind="lib" path="tools/lib/joda-time-2.8.1.jar"/>
```
Comment (yamelsenih) [inline .classpath:317]: Please remove this change and create a new pull request only for this. Is more clean

File: base/build.gradle [hunk 3]
```diff
@@ -85,6 +85,13 @@ dependencies {
     //api 'io.github.adempiere:tools:'
 }
 
+configurations.all {
```
Comment (yamelsenih) [inline base/build.gradle:88]: Please remove this change and create a new pull request only for this. Is more clean

**Review by yamelsenih — CHANGES_REQUESTED** [review]
Please change this and reply this comment

File: base/src/org/compiere/model/MAttachment.java [hunk 4]
```diff
@@ -1026,7 +1043,7 @@ public boolean updateEntry(int i, File file)
 		}
 		log.fine("updateEntry - " + file);
 		//
-		String name = file.getName();
+		// String name = file.getName();
```
Comment (yamelsenih) [inline base/src/org/compiere/model/MAttachment.java:1046]: Remove this comment if is unused

File: migration/394lts-395lts/09940_4176_File_Handler_not_supported_on_System_client.xml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<Migrations>
+  <Migration EntityType="D" Name="#4176 File Handler not supported on System client" ReleaseNo="3.9.5" SeqNo="9940">
```
Comment (yamelsenih) [inline migration/394lts-395lts/09940_4176_File_Handler_not_supported_on_System_client.xml:3]: Please change the ReleaseNo="3.9.5" to ReleaseNo="3.9.4.001"

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: migration/394lts-395lts/09940_4176_File_Handler_not_supported_on_System_client.xml (hunk 5 above)
Comment (EdwinBetanc0urt) [reply migration/394lts-395lts/09940_4176_File_Handler_not_supported_on_System_client.xml:3]: All xml files on this folder are tagged with release 3.9.5 https://github.com/adempiere/adempiere/tree/hotfix/3.9.4.001/migration/394lts-395lts

File: base/src/org/compiere/model/MAttachment.java (hunk 4 above)
Comment (EdwinBetanc0urt) [reply base/src/org/compiere/model/MAttachment.java]: changes implemented

File: .classpath (hunk 2 above)
Comment (EdwinBetanc0urt) [reply .classpath:317]: These changes are necessary for the file handler to work.

File: base/build.gradle (hunk 3 above)
Comment (EdwinBetanc0urt) [reply base/build.gradle:89]: These changes are necessary for the file handler to work.

File: .classpath (hunk 2 above)
Comment (yamelsenih) [reply .classpath:317]: Please make other pull request with these changes

File: base/build.gradle (hunk 3 above)
Comment (yamelsenih) [reply base/build.gradle:89]: Make other pull request with these changes

**Review by yamelsenih — CHANGES_REQUESTED** [review]

File: base/build.gradle (hunk 3 above)
Comment (EdwinBetanc0urt) [reply base/build.gradle]: The requested changes were implemented

File: .classpath (hunk 2 above)
Comment (EdwinBetanc0urt) [reply .classpath]: The requested changes were implemented

File: migration/394lts-395lts/09940_4176_File_Handler_not_supported_on_System_client.xml (hunk 5 above)
Comment (EdwinBetanc0urt) [reply migration/394lts-395lts/09940_4176_File_Handler_not_supported_on_System_client.xml]: The xml migration file are change to 3.9.4.001 release

File: base/build.gradle (hunk 3 above)
Comment (EdwinBetanc0urt) [reply base/build.gradle]: The requested changes were implemented

**Review by yamelsenih — APPROVED** [review]

**Pull request merged**